### PR TITLE
fix: Profile listings correction

### PR DESCRIPTION
**Pull request by flesher1**
Corrected issues where:
- Users can see deleted listings on profile pages
- Users can see un-approved/disabled listings or other users profile pages
- Modified test file to test for crrect case where the listing is deleted or not
https://github.com/aml-development/ozp-center/issues/865

### CODE REVIEW

**Comment by tdressel** [comment]
:+1:  Pulled branch and verified the attached test case.

File: app/js/components/profile/ProfileWindow.jsx [hunk 1]
```diff
@@ -123,9 +123,14 @@ var ProfileInfo = React.createClass({
     render: function() {
         var profile = this.state.profile,
             linkEl = this.props.listingLinkEl,
-            listings = this.state.ownedListings.map(
-                l => <ListingRow key={l.id} linkEl={linkEl} listing={l} />
-            );
+            listings = this.state.ownedListings.map(function(l){
+                if (CurrentProfileStore.isSelf && !l.isDeleted){
+                    return <ListingRow key={l.id} linkEl={linkEl} listing={l} />;
+                }else if (l.isEnabled && !l.isDeleted && (l.approvalStatus == "APPROVED")){
```
Comment (blynch11p) [inline app/js/components/profile/ProfileWindow.jsx]: The return value is the same for both if and else, so, why not just make a single `if`?
something like: 
```
if(!l.isDeleted && (CurrentProfileStore.isSelf || (l.isEnabled && (l.approvalStatus == "APPROVED")))
```

**Comment by blynch11p** [comment]
Other than looking at my comment in code, functionality is good 👍

File: app/js/components/profile/ProfileWindow.jsx (hunk 1 above)
Comment (flesher1) [reply app/js/components/profile/ProfileWindow.jsx]: For readability and future possible changes, I would like to keep this the way it is.

Comment (blynch11p) [reply app/js/components/profile/ProfileWindow.jsx]: That makes sense, and as we discussed yesterday, it allows for quick changes (ie: removing isDeleted from CurrentProfileStore is we want)

**Pull request merged**